sklearn/metrics/_classification.py

```diff
@@ -2285,6 +2285,48 @@ def log_loss(y_true, y_pred, *, eps=1e-15, normalize=True, sample_weight=None,
     return _weighted_sum(loss, sample_weight, normalize)
 
 
+def _check_valid_multiclass_decision_shape(y_true_unique, pred_decision,
+                                           labels):
+    """Check if the inputs are consistent.
+
+    If it is a multiclass problem (it means thare are more than 2 classes
+    to classify), `pred_decision` parameter can't be 1d array. The column
+    number must be equal to classes number.
+
+    If not every classes are present in `y_true`, `labels` parameter
+    must be filled in.
+
+    Parameters
+    ----------
+    y_true_unique : array-like
+
+    pred_decision : array-like
+
+    labels : array-like
+    """
+
+    if pred_decision.ndim <= 1:
+        raise ValueError("The shape of pred_decision can not be 1d array"
+                         "with a multiclass target. "
+                         "pred_decision shape must be (n_samples, n_classes)")
+
+    invalid_decision_shape = (pred_decision.ndim > 1 and
+                              y_true_unique.size != pred_decision.shape[1])
+
+    if labels is None:
+        if invalid_decision_shape:
+            raise ValueError("Please include all labels in y_true "
+                             "or pass labels as third argument")
+    elif invalid_decision_shape:
+        raise ValueError("The shape of pred_decision is not "
```

Comment:
With the `pred_decision.ndim <= 1` handled before hand. I think the following is simple enough to inlined into `hinge_loss`.

```python
if pred_decision.ndim <= 1:
	raise ValueError("The shape of pred_decision can not be 1d...")

# pred_decision.ndim > 1 is true
if y_true_unique.size != pred_decision.shape[1]:
	if labels is None:
		raise ValueError("Please include all...")
	else:
		raise ValueError("The shape of pred_decision...")
```

Thank you for your patience on this PR @PierreAttard .

Reply:
No, it's ok !